Add test plans for lecturer interface (#114)
Gamify-IT/docs: dev-manuals/test-plans/lecturer-interface-frontend/plans/u-lecturer-interface-15-edit-npc-configuration.md

# Lecturer-Interface: Edit npc configuration (`u.lecturer-interface-15`)

Version: V1.0, 10.08.2022 \
Author: Max Kästner

## Description

The configuration of a npc can be edited in a modal. For npcs of a world the path is `/courses/<course-id>/worlds/<world-index>/npcs` and for a dungeon `/courses/<course-id>/worlds/<world-index>/dungeons/<dungeon-index>/npcs`.

## Precondition

> At least one course is created. The tester is on the view of an area npcs list as described in the description above.
> 
> ## Postcondition
> 
> The configuration of a selected npc was updated.
> 
> ## Typical procedure
> 
> 1. Click the edit button of a npc
> 2. A modal opens
> 3. Press the `Add Input` button
> 4. A new text field appears
> 5. Enter a little sentence a npc could say
> 6. Press the Ok button of the modal
> 7. The modal closes
> 8. A feedback toast message appears which contains that configuraiton was saved
> 9. Press the edit button from the npc again and make sure the configurations are the same as you saved it before.
> 
> ## Alternative procedures
> 
> 6.1: Press the Cancel button of the modal: \
>     6.1.1 The modal closes \
>     6.1.2 Press the edit button from the npc again and make sure the configurations that you changed before cancelling were not applied.

Reviewer: ```suggestion
At least one course is created. The tester is on the view listing the NPCs of an area.

## Postcondition

The configuration of a selected NPC was updated.

## Typical procedure

1. Click the edit button of the NPC
2. A modal opens
3. Press the `Add Input` button
4. A new text field appears
5. Enter a little text the NPC should say
6. Press the Ok button of the modal
7. The modal closes
8. A feedback toast message appears which contains that configuration was saved

## Alternative procedures

6.1.1. Press the Cancel button of the modal to cancel editing and to close the modal
```

Author: Is it not important that you check that the system does really what it should do?

Reviewer: Why do you mean?
Because I removed the part that you should restore the previous state?
I trust our testers to be smart enough to consider that already when testing, or to leave it be if they don't care about changing the values in the database.

Or what else do you mean? I'm a little confused by that question...

Author: ```suggestion
At least one course is created. The tester is on the view of an area NPCs list as described in the description above.

## Postcondition

The configuration of a selected NPC was updated.

## Typical procedure

1. Click the edit button of a NPC
2. A modal opens
3. Press the `Add Input` button
4. A new text field appears
5. Enter a little sentence a NPC could say
6. Press the Ok button of the modal
7. The modal closes
8. A feedback toast message appears which contains that configuraiton was saved
9. Press the edit button from the NPC again and ensure that the configuration is what you saved above.

## Alternative procedures

6.1: Press the Cancel button of the modal: \
    6.1.1 The modal closes \
    6.1.2 Press the edit button from the NPC again and ensure that the configuration has not changed.
```
